include/irr/core/math/aabb2dSIMD.h

```diff
@@ -8,90 +8,92 @@
 #include <iostream>
 
 #define SIGN_FLIP_MASK_XY	core::vector3du32_SIMD(0x80000000, 0x80000000, 0x00000000, 0x00000000)
-#define SIGN_FLIP_MASK_XYZW core::vector3du32_SIMD(0x80000000, 0x80000000, 0x80000000, 0x80000000)
-
 
 namespace irr { namespace core {
 
+
 class aabb2dSIMDf
 {
 public:
 	inline aabb2dSIMDf()
-		: box( FLT_MAX, FLT_MAX, FLT_MAX, FLT_MAX ) {};
+		: internalBoxRepresentation(-FLT_MAX, -FLT_MAX, -FLT_MAX, -FLT_MAX) {};
 
-	inline aabb2dSIMDf(const core::vectorSIMDf& _box)
-		: box(_box ^ SIGN_FLIP_MASK_XY) {};
+	inline explicit aabb2dSIMDf(const core::vectorSIMDf& _box)
+		: internalBoxRepresentation(_box^ SIGN_FLIP_MASK_XY) {};
 
-	inline void addBox(const aabb2dSIMDf& _box)		
-	{ 
-		box = core::max_(box, _box.getBounds()); 
+	inline void addBox(const aabb2dSIMDf& _box)
+	{
+		internalBoxRepresentation = core::max_(internalBoxRepresentation, _box.getBounds());
 	}
 
-	inline void addPoint(const core::vectorSIMDf& point)	
+	inline void addPoint(const core::vectorSIMDf& point)
 	{
-		addBox(point); 
+		addBox(aabb2dSIMDf(point));
```

Comment:
this sort of thing could be provided by a base class with CRTP (assuming the base class a CRTP template argument which is a template class itself, see the allocator code for an example of a template on a template class)
```cpp
inline void addPoint(const T& point)
{
   CRTP<T>::addPoint(CRTP<T>(point));
}
```

Reply:
another note, it would be good to use the `vector2df_SIMD` typedef for places where we input or return points, its more clear what the expected input and output is